src/worker.py

```
@@ -976,13 +976,21 @@ async def api_join(req, env):
+    ).bind(act_id, user["id"]).first()
+    if existing:
+        return ok(None, "Already joined activity")
+
     enr_id = new_id()
     try:
         await env.DB.prepare(
-            "INSERT OR IGNORE INTO enrollments (id,activity_id,user_id,role)"
+            "INSERT INTO enrollments (id,activity_id,user_id,role)"
             " VALUES (?,?,?,?)"
         ).bind(enr_id, act_id, user["id"], role).run()
     except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)